File: CS-Examples/10_Shapes/SetShapeOrder.cs
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet.Shapes;
using System;
using System.Windows.Forms;


namespace SetShapeOrder
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook wb = new Workbook();
            //Load an excel file
            wb.LoadFromFile(@"..\..\..\..\..\..\Data\SetShapeOrder.xlsx");

            //Bring the picture forward one level
            wb.Worksheets[0].Pictures[0].ChangeLayer(ShapeLayerChangeType.BringForward);

            //Bring the image in fron of all other objects
            wb.Worksheets[1].Pictures[0].ChangeLayer(ShapeLayerChangeType.BringToFront);

            //Send the shape back one level
            XlsShape shape = wb.Worksheets[2].PrstGeomShapes[1] as XlsShape;
            shape.ChangeLayer(ShapeLayerChangeType.SendBackward);

            //Send the shape behind all other objects
            shape = wb.Worksheets[3].PrstGeomShapes[1] as XlsShape;
            shape.ChangeLayer(ShapeLayerChangeType.SendToBack);

            String result = "SetShapeOrder_result.xlsx";
            //Save to file
            wb.SaveToFile(result, ExcelVersion.Version2010);
            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: VB-Examples/10_Shapes/SetShapeOrder.vb
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet.Shapes


Namespace SetShapeOrder
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim wb As New Workbook()

            ' Load an existing Excel file into the workbook
            wb.LoadFromFile("..\..\..\..\..\..\Data\SetShapeOrder.xlsx")

            ' Bring the first picture in the first worksheet forward by changing its layer
            wb.Worksheets(0).Pictures(0).ChangeLayer(ShapeLayerChangeType.BringForward)

            ' Bring the first picture in the second worksheet to the front by changing its layer
            wb.Worksheets(1).Pictures(0).ChangeLayer(ShapeLayerChangeType.BringToFront)

            ' Get the first PrstGeomShape in the third worksheet and cast it to XlsShape type
            Dim shape As XlsShape = TryCast(wb.Worksheets(2).PrstGeomShapes(1), XlsShape)
            ' Send the shape backward by changing its layer
            shape.ChangeLayer(ShapeLayerChangeType.SendBackward)

            ' Get the first PrstGeomShape in the fourth worksheet and cast it to XlsShape type
            shape = TryCast(wb.Worksheets(3).PrstGeomShapes(1), XlsShape)
            ' Send the shape to the back by changing its layer
            shape.ChangeLayer(ShapeLayerChangeType.SendToBack)

            ' Specify the file name for the resulting Excel file
            Dim result As String = "SetShapeOrder_result.xlsx"

            ' Save the workbook to the specified file in Excel 2010 format
            wb.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            wb.Dispose()
            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
